### PR TITLE
lerpn: init at unstable-2023-06-08

### PR DESCRIPTION
**Pull request by ckiee**
###### Description of changes

https://gitea.alexisvl.rocks/alexisvl/lerpn

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`, tested {aarch64,x86_64}-linux)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/lerpn/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchFromGitea {
+    domain = "gitea.alexisvl.rocks";
+    owner = "alexisvl";
+    repo = pname;
+    rev = "df792ae930005415afd4e56950de671cdd23fb26";
+    sha256 = "sha256-XfQssik8POBgfCbmP3tjUGomE769yrujqJhImd6vxTQ=";
+  };
+
+  dontUseSetuptoolsCheck = true; # doesn't halt
+
```
Comment (natsukium) [inline pkgs/tools/misc/lerpn/default.nix:16]: Would you please add `checkPhase`?
The upstream has a test script.

and it is preferable to add `pythonImportsCheck` for a quick check.

File: pkgs/tools/misc/lerpn/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  meta = with lib; {
+    homepage = "https://gitea.alexisvl.rocks/alexisvl/lerpn";
+    description = "Curses RPN calculator written in straight Python";
+    maintainers = with maintainers; [ ckie ];
+    license = licenses.gpl3;
```
Comment (natsukium) [inline pkgs/tools/misc/lerpn/default.nix:21]:
```suggestion
    license = licenses.gpl3Plus;
```
see
https://gitea.alexisvl.rocks/alexisvl/lerpn/src/commit/b65e56cfbbb38f8200e7b0c18b3a585ae768c6e2/LerpnApp/__main__.py#L5-L6

File: pkgs/tools/misc/lerpn/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+python3.pkgs.buildPythonApplication rec {
+  pname = "lerpn";
+  version = "unstable-2023-06-08";
+
+  src = fetchFromGitea {
+    domain = "gitea.alexisvl.rocks";
+    owner = "alexisvl";
+    repo = pname;
+    rev = "df792ae930005415afd4e56950de671cdd23fb26";
+    sha256 = "sha256-XfQssik8POBgfCbmP3tjUGomE769yrujqJhImd6vxTQ=";
```
Comment (natsukium) [inline pkgs/tools/misc/lerpn/default.nix:12]:
```suggestion
    hash = "sha256-XfQssik8POBgfCbmP3tjUGomE769yrujqJhImd6vxTQ=";
```

File: pkgs/tools/misc/lerpn/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{ python3, lib, fetchFromGitea }:
```
Comment (natsukium) [inline pkgs/tools/misc/lerpn/default.nix:1]: One item per line is preferred.

File: pkgs/tools/misc/lerpn/default.nix (hunk 2 above)
Comment (ckiee) [reply pkgs/tools/misc/lerpn/default.nix:21]: ah, missed that.

File: pkgs/tools/misc/lerpn/default.nix (hunk 4 above)
Comment (ckiee) [reply pkgs/tools/misc/lerpn/default.nix:1]: Reasoning? nixpkgs-fmt is happy.

File: pkgs/tools/misc/lerpn/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "lerpn";
+  version = "unstable-2023-06-09";
+
+  src = fetchFromGitea {
+    domain = "gitea.alexisvl.rocks";
+    owner = "alexisvl";
+    repo = pname;
+    rev = "b65e56cfbbb38f8200e7b0c18b3a585ae768c6e2";
```
Comment (ckiee) [inline pkgs/tools/misc/lerpn/default.nix:11]: I also bumped it, there was a bugfix upstream.

File: pkgs/tools/misc/lerpn/default.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/tools/misc/lerpn/default.nix:1]: Future diffs will be better-looking.
Adding a new input will add a new green line, removing an input will generate a new red line.
On the other hand, your method will add a new green and a new red line at each new modification.

File: pkgs/tools/misc/lerpn/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{ python3, lib, fetchFromGitea }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "lerpn";
+  version = "unstable-2023-06-09";
+
+  src = fetchFromGitea {
+    domain = "gitea.alexisvl.rocks";
+    owner = "alexisvl";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/tools/misc/lerpn/default.nix:10]:
```suggestion
    repo = "lerpn";
```

No reason to do this besides overusing `rec`.

File: pkgs/tools/misc/lerpn/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "b65e56cfbbb38f8200e7b0c18b3a585ae768c6e2";
+    hash = "sha256-4xqBHcOWHAvQtXS9CJWTGTdE4SGHxjghZY+/KPUgX70=";
+  };
+
+  checkPhase = ''
```
Comment (AndersonTorres) [inline pkgs/tools/misc/lerpn/default.nix:15]: Add `runHook`s.

File: pkgs/tools/misc/lerpn/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
+{ python3, lib, fetchFromGitea }:
+
+python3.pkgs.buildPythonApplication rec {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/lerpn/default.nix:3]:
```suggestion
python3.pkgs.buildPythonApplication {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/lerpn/default.nix (hunk 4 above)
Comment (ckiee) [reply pkgs/tools/misc/lerpn/default.nix]: Ah this makes a lot of sense after seeing `nixpkgs-fmt` putting the commas at the *start* of the lines when I do this, too.

Thank you! It seems I'm a little rusty, but also [I've never had you before](https://github.com/NixOS/nixpkgs/pulls?q=reviewed-by%3AAndersonTorres+author%3Ackiee+), so hi! :P

(feel free to resolve this, leaving it open for visbility)

**Comment by ckiee** [comment]
Ready for re-review. Still builds on x86_64-linux, unsurprisingly.

**Review by AndersonTorres — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
It works fine on `aarch64-darwin` too.

Thanks for your help, @AndersonTorres san.

**Pull request merged**